src/utils/helpers/browser/browser.js

```diff
@@ -89,6 +98,16 @@ const Browser = {
   },
 
   /**
+   * Sets the body scroll
+   *
+   * @param {Boolean} enabled
+   * @method setBodyScroll
+   */
+  setBodyScroll: (enabled) => {
+    Browser.getBody().style.overflow = enabled ? '' : 'hidden';
```

Comment:
You might be able to add something like this to `dialog.scss`:

```
.carbon-dialog__dialog--open {
  body & {
    overflow: hidden;
  }
}
```

And add this line to the object returned by the `dialogClasses` getter:

`'carbon-dialog__dialog--open': this.props.open`